SQL_Assignment_day3_A.sql

```diff
@@ -0,0 +1,101 @@
+    DoctorName VARCHAR(100),
+    Specialty VARCHAR(100)
+);
+
+CREATE TABLE Appointments (
+    AppointmentID INT PRIMARY KEY,
+    PatientID INT,
+    DoctorID INT,
+    AppointmentDate DATE,
+    FOREIGN KEY (PatientID) REFERENCES Patients(PatientID),
+    FOREIGN KEY (DoctorID) REFERENCES Doctors(DoctorID)
+);
+
+CREATE TABLE Medications (
+    MedicationID INT PRIMARY KEY,
+    MedicationName VARCHAR(200),
+    DosageForm VARCHAR(50)
+);
+
+CREATE TABLE Prescriptions (
+    PrescriptionID INT PRIMARY KEY,
+    PatientID INT,
+    MedicationID INT,
+    PrescriptionDate DATE,
+    FOREIGN KEY (PatientID) REFERENCES Patients(PatientID),
+    FOREIGN KEY (MedicationID) REFERENCES Medications(MedicationID)
+);
+
+-- Questions
+
+-- 1. List all patients and their appointments, including patients who have never had an appointment.
+SELECT PATIENTS.PATIENTNAME, APPOINTMENTS.APPOINTMENTID 
+FROM PATIENTS LEFT JOIN APPOINTMENTS
+ON PATIENTS.PATIENTID=APPOINTMENTS.PATIENTID;
+
+-- 2. Display all doctors and their scheduled appointments, including doctors without any appointments.
+SELECT DOCTORS.DOCTORNAME, APPOINTMENTS.APPOINTMENTID FROM DOCTORS
+LEFT JOIN  APPOINTMENTS ON APPOINTMENTS.DOCTORID=DOCTORS.DOCTORID;
+
+-- 3. Show all medications and the patients they've been prescribed to, including medications that haven't been prescribed.
+SELECT PRESCRIPTIONS.PRESCRIPTIONID, PATIENTS.PATIENTNAME
+FROM PRESCRIPTIONS LEFT JOIN PATIENTS ON PRESCRIPTIONS.PATIENTID=PATIENTS.PATIENTID;
+-- 4. List all possible patient-doctor combinations, regardless of whether an appointment has occurred.
+SELECT DOCTORS.DOCTORNAME, PATIENTS.PATIENTNAME
+FROM DOCTORS CROSS JOIN PATIENTS;
+
+-- 5. Display all prescriptions with patient and medication information, including prescriptions where either the patient or medication information is missing.
+SELECT 
+    P.PrescriptionID,
+    Pat.PatientName,
+    Med.MedicationName,
+    P.PrescriptionDate
+FROM 
+    Prescriptions P
+LEFT JOIN 
+    Patients Pat ON P.PatientID = Pat.PatientID
+LEFT JOIN 
+    Medications Med ON P.MedicationID = Med.MedicationID;
+-- 6. Show all patients who have never been prescribed any medication, along with their appointment history.
+SELECT PAT.PATIENTID, PAT.PATIENTNAME, DOC.DOCTORID
+FROM PATIENTS PAT LEFT JOIN PRESCRIPTIONS PRESC ON PAT.PATIENTID=PRESC.PATIENTID
+LEFT JOIN  APPOINTMENTS APP ON PAT.PATIENTID=APP.PATIENTID
+LEFT JOIN DOCTORS DOC ON DOC.DOCTORID=APP.DOCTORID
+
+-- 7. List all doctors who have appointments in the next week, along with the patients they're scheduled to see.
+SELECT DOC.DOCTORNAME, APP.APPOINTMENTDATE
+FROM APPOINTMENTS APP LEFT JOIN DOCTORS DOC ON APP.DOCTORID=DOC.DOCTORID
+LEFT JOIN PATIENTS PAT ON PAT.PATIENTID=APP.PATIENTID
+WHERE APP.APPOINTMENTDATE BETWEEN GETDATE() AND DATEADD(DAY,7,GETDATE());
+
+
+-- 8. Display all medications prescribed to patients over 60 years old, including medications not prescribed to this age group.
+FROM Medications Med
```

Comment:
The query has a syntax error; it lacks a SELECT statement at the beginning. Additionally, ensure proper calculation for age (it should use DATEDIFF correctly).

Reply:
Corrections made